### PR TITLE
SOLR-16929 SolrStream propagates undecoded error message

### PR DESCRIPTION
**Pull request by stillalex**
https://issues.apache.org/jira/browse/SOLR-16929

# Description

I am proposing to only log the error as string in the cases of auth exceptions. in the other cases the previous behavior is restored. I also added missing support for error response in the case of /select handler with javabin format.

# Solution

Please provide a short description of the approach taken to implement your solution.

# Tests

Please describe the tests you've developed or run to confirm this patch implements the feature or solves the problem.

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `main` branch.
- [x] I have run `./gradlew check`.
- [x] I have added tests for my changes.
- [x] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

**Review by risdenk — APPROVED** [review]

File: solr/solrj-streaming/src/test/org/apache/solr/client/solrj/io/stream/StreamingTest.java [hunk 1]
```diff
@@ -627,16 +627,15 @@ public void testParallelReducerStream() throws Exception {
...
   public void testExceptionStream() throws Exception {
 
     helloDocsUpdateRequest.commit(cluster.getSolrClient(), COLLECTIONORALIAS);
 
     StreamContext streamContext = new StreamContext();
     SolrClientCache solrClientCache = new SolrClientCache();
     streamContext.setSolrClientCache(solrClientCache);
-    // Test an error that comes originates from the /select handler
     try {
+      // Test an error that comes originates from the /select handler
```
Comment (epugh) [inline solr/solrj-streaming/src/test/org/apache/solr/client/solrj/io/stream/StreamingTest.java:638]: "that originates"

File: solr/solrj-streaming/src/test/org/apache/solr/client/solrj/io/stream/StreamingTest.java [hunk 2]
```diff
@@ -646,16 +645,45 @@ public void testExceptionStream() throws Exception {
...
+      sParamsA = params("q", "*:*", "fl", "a_s,a_i,a_f,blah", "sort", "blah asc", "wt", "javabin");
+      stream = new CloudSolrStream(zkHost, COLLECTIONORALIAS, sParamsA);
+      estream = new ExceptionStream(stream);
+      estream.setStreamContext(streamContext);
+      t = getTuple(estream);
+      assertTrue(t.EOF);
+      assertTrue(t.EXCEPTION);
+      assertTrue(t.getException().contains("sort param field can't be found: blah"));
+
+      // Test an error that comes originates from the /export handler
```
Comment (epugh) [inline solr/solrj-streaming/src/test/org/apache/solr/client/solrj/io/stream/StreamingTest.java:657]: "that originates"

File: solr/solrj-streaming/src/test/org/apache/solr/client/solrj/io/stream/StreamingTest.java [hunk 3]
```diff
@@ -646,16 +645,45 @@ public void testExceptionStream() throws Exception {
...
+      stream = new CloudSolrStream(zkHost, COLLECTIONORALIAS, sParamsA);
+      estream = new ExceptionStream(stream);
+      estream.setStreamContext(streamContext);
+      t = getTuple(estream);
+      assertTrue(t.EOF);
+      assertTrue(t.EXCEPTION);
+      assertTrue(t.getException().contains("sort param field can't be found: blah"));
+
       // Test an error that comes originates from the /export handler
-      sParamsA = params("q", "*:*", "fl", "a_s,a_i,a_f,score", "sort", "a_s asc", "qt", "/export");
```
Comment (epugh) [inline solr/solrj-streaming/src/test/org/apache/solr/client/solrj/io/stream/StreamingTest.java:650]: "that originates" ;-)

**Review by epugh — APPROVED** [review]
LGTM.  I'm not super deep in the underlying "how all this works", but from looking, it makes sense.   A couple of comment language tweaks ;-)

**Comment by stillalex** [comment]
thanks @epugh I usually pay more attention to these details, but this time I missed it. I want to blame the previous code for this :)

**Comment by stillalex** [comment]
Raised the backwards compatibility aspect of this change on the Solr Dev list https://lists.apache.org/thread/zqqgofwrqd6xw3x139zr08loofzopsbb

**Pull request merged**